### PR TITLE
Add train.py script for local model training

### PR DESCRIPTION
**Pull request by the-sink**
This script is essentially a clone of the training code in the colab instance, but with a few variables pushed to the top and the addition of a `use_gpu` variable. Setting it to false disables `use_cuda` and `use_multiprocessing`, making training very slow, but it should allow many more people to use local training if they'd like. Also edited the bot label to pull the current date/time automatically and append the local time to the end. Addresses #20.

### CODE REVIEW

**Review by zacc — CHANGES_REQUESTED** [review]

File: model_finetuning/train.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+model_type = "gpt2"
+# IF USING GPT2: for the small model, use just "gpt2", otherwise append "-medium"/"-large"/"-xl" depending on the model size you'd like to use
+# anything above small is very unlikely to run unless you have a whole lot of memory and a beefy GPU - even the colab instance usually can't do medium
+model_name = "gpt2"
+
+# if set to False, training will occur exclusively on the CPU without multiprocessing (this will be EXTREMELY slow but will work on more systems)
+use_gpu = False
+
+
+currentDateTime = datetime.now()
```
[the remark this answers is not in the document]
Comment (the-sink) [reply model_finetuning/train.py]: Replaced with `current_date_time` (27e691a)

File: model_finetuning/train.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "gradient_accumulation_steps": 1,
+    "max_steps": 12000,
+
+    # Use text because of grouping by reddit submission
+    "dataset_type": "simple",
+    # Sliding window will help it manage very long bits of text in memory
+    "sliding_window": True,
+    "max_seq_length": 512,
+
+		"mlm": False, # has to be false for gpt-2
```
[the remark this answers is not in the document]
Comment (the-sink) [reply model_finetuning/train.py]: Fixed (27e691a)

File: model_finetuning/train.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    # Save disk space by only saving on checkpoints
+    "save_eval_checkpoints": True,
+    "save_model_every_epoch": False,
+    # disable saving each step to save disk space
+    "save_steps": -1, 
+
+    "use_multiprocessing": use_gpu, # set to false when not using a gpu due to instability (often throws RuntimeErrors)
+
+    "output_dir": f"{bot_label}/",
+		"best_model_dir": f"{bot_label}/best_model",
```
Comment (zacc) [inline model_finetuning/train.py:54]: Sorry, we missed this indent!

Comment (the-sink) [reply model_finetuning/train.py]: Whoops, fixed 1835a94

**Comment by zacc** [comment]
I didn't try and run it locally but the code looks right :shrug:

**Pull request merged**